spinalcordtoolbox/scripts/sct_register_to_template.py

```diff
@@ -456,6 +456,21 @@ def main(argv: Sequence[str]):
     out.data = binarize(out.data, 0.5)
     out.save(path=ftmp_seg)
 
+    # Change orientation of input images to RPI
+    printv('\nChange orientation of input images to RPI...', verbose)
+
+    img_tmp_data = Image(ftmp_data).change_orientation("RPI")
+    ftmp_data = add_suffix(img_tmp_data.absolutepath, "_rpi")
+    img_tmp_data.save(path=ftmp_data, mutable=True)
+
+    img_tmp_seg = Image(ftmp_seg).change_orientation("RPI")
+    ftmp_seg = add_suffix(img_tmp_seg.absolutepath, "_rpi")
+    img_tmp_seg.save(path=ftmp_seg, mutable=True)
+
+    img_tmp_label = Image(ftmp_label).change_orientation("RPI")
+    ftmp_label = add_suffix(img_tmp_label.absolutepath, "_rpi")
+    img_tmp_label.save(ftmp_label, mutable=True)
```

Comment:
It looks like these changes have caused deviations in the batch_processing.sh result:

```python
 =========================== short test summary info ============================
FAILED testing/batch_processing/test_batch_processing.py::test_batch_processing_results[mt/mtr_in_wm.csv-0-MAP()] - assert 54.3539623020136 == 54.39462068731342 ± 5.4e-05
  comparison failed
  Obtained: 54.3539623020136
  Expected: 54.39462068731342 ± 5.4e-05
FAILED testing/batch_processing/test_batch_processing.py::test_batch_processing_results[dmri/fa_in_cst.csv-0-WA()] - assert 0.7746611002178296 == 0.7793857115174943 ± 7.8e-07
  comparison failed
  Obtained: 0.7746611002178296
  Expected: 0.7793857115174943 ± 7.8e-07
FAILED testing/batch_processing/test_batch_processing.py::test_batch_processing_results[dmri/fa_in_cst.csv-1-WA()] - assert 0.7727014657926683 == 0.7756883298906552 ± 7.8e-07
  comparison failed
  Obtained: 0.7727014657926683
  Expected: 0.7756883298906552 ± 7.8e-07
```

This is very curious to me! I didn't realize that there would be a meaningful difference between (Reorientation -> Resampling) vs. (Resampling -> Reorientation).

I could instead scope the changes to be even more narrow (e.g. adding a reorientation step to _just_ the `resample_labels` function) which may minimize the numerical impact of these changes.

But, from a code organization perspective, it does feel nice to deduplicate the reorientation...

Reply:
I'm not totally surprised to see that it would make a noticeable change. I guess the resampling grid is always aligned with one corner of the image box, and each dimension may be a fractional multiple of the grid step. In that case, reorienting the image means picking a different corner of the image box to align the resampling grid, and that can be a substantial change.

But because of this, I think it would be a good idea to _always_ reorient before resampling. That would mean that we have a consistent choice of resampling grid alignment, no matter what the input orientation is.

Since SCT works hard to make orientation usually not matter, I think users would be surprised to find out that this seemingly invisible property of the image file actually affects the result of an operation like resampling.

Reply:
As discussed in the most recent SCT meeting, we will keep the code changes here and update the batch processing results accordingly.